src/main/java/net/dv8tion/jda/bot/sharding/DefaultShardManagerBuilder.java

```diff
@@ -0,0 +1,546 @@
+/*
+ *     Copyright 2015-2017 Austin Keener & Michael Ritter & Florian Spieß
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package net.dv8tion.jda.bot.sharding;
+
+import com.neovisionaries.ws.client.WebSocketFactory;
+import java.util.*;
+import java.util.stream.Collectors;
+import javax.security.auth.login.LoginException;
+import net.dv8tion.jda.bot.entities.impl.DefaultShardManagerImpl;
+import net.dv8tion.jda.core.OnlineStatus;
+import net.dv8tion.jda.core.audio.factory.IAudioSendFactory;
+import net.dv8tion.jda.core.entities.Game;
+import net.dv8tion.jda.core.exceptions.RateLimitedException;
+import net.dv8tion.jda.core.hooks.IEventManager;
+import net.dv8tion.jda.core.requests.SessionReconnectQueue;
+import net.dv8tion.jda.core.utils.Checks;
+import okhttp3.OkHttpClient;
+
+/**
+ * Used to create new {@link net.dv8tion.jda.bot.entities.impl.DefaultShardManagerImpl DefaultShardManagerImpl} instances.
+ *
+ * <p>A single DefaultShardManagerBuilder can be reused multiple times. Each call to {@link #buildAsync()} or {@link #buildBlocking()}
+ * creates a new {@link net.dv8tion.jda.bot.entities.impl.DefaultShardManagerImpl DefaultShardManagerImpl} instance using the same information.
+ *
+ * @since  3.1
+ * @author Aljoscha Grebe
+ */
+public class DefaultShardManagerBuilder
+{
+    protected Collection<Integer> shards = null;
+    protected int shardsTotal = 1;
+    protected IAudioSendFactory audioSendFactory = null;
+    protected OkHttpClient.Builder httpClientBuilder = null;
+    protected WebSocketFactory wsFactory = null;
+    protected boolean autoReconnect = true;
+    protected int backoff = 200;
+    protected int corePoolSize = 2;
+    protected boolean enableBulkDeleteSplitting = true;
+    protected boolean enableShutdownHook = true;
+    protected boolean enableVoice = true;
+    protected IEventManager eventManager = null;
+    protected Game game = null;
+    protected boolean idle = false;
+    protected final List<Object> listeners = new ArrayList<>();
+    protected int maxReconnectDelay = 900;
+    protected OnlineStatus status = OnlineStatus.ONLINE;
+    protected String token = null;
+    private SessionReconnectQueue reconnectQueue;
```

Comment:
Group these fields by type such as `boolean` or `int` to give it a better structure.
Why is reconnectQueue the only `private` field here?